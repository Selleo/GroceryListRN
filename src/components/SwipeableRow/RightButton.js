import * as React from 'react'
import { Animated, StyleSheet } from 'react-native'
import { RectButton } from 'react-native-gesture-handler'
import Icon from 'react-native-vector-icons/FontAwesome'

import { white } from '../../styles/colors'

type Props = {
  RTL: boolean,
  backgroundColor: string,
  iconName: string,
  onPress: Function,
  positionX: number,
  progress: object<{ interpolate: Function }>,
}

export default class RightButton extends React.PureComponent<Props> {
  render() {
    const { progress, positionX, iconName, onPress, backgroundColor, RTL } = this.props
    const outputRange = RTL ? [-positionX, 0] : [positionX, 0]

    const trans = progress.interpolate({
      inputRange: [0, 1],
      outputRange,
    })

    return (
      <Animated.View style={[styles.container, { transform: [{ translateX: trans }] }]}>
        <RectButton onPress={onPress} style={[styles.rightAction, { backgroundColor }]}>
          <Icon color={white} name={iconName} size={24} />
        </RectButton>
      </Animated.View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  rightAction: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
})
