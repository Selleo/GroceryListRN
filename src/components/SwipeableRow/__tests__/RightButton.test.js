import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import RightButton from '../RightButton'

describe('<RightButton />', () => {
  const defaultProps = {
    backgroundColor: '#000',
    iconName: 'music',
    onPress: jest.fn(),
    positionX: 1,
    progress: { interpolate: jest.fn() },
  }

  const wrapper = renderer.create(<RightButton {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
