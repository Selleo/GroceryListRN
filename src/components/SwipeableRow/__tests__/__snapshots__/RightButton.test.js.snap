// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightButton /> render 1`] = `
<RCTView
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "transform": Array [
          Object {
            "translateX": undefined,
          },
        ],
      },
    ]
  }
>
  <RNGestureHandlerButton
    activeOpacity={0.105}
    collapsable={false}
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={[MockFunction]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#000",
          },
        ],
      ]
    }
    underlayColor="black"
  >
    <RCTView
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "opacity": Object {
              "__makeNative": [MockFunction],
              "addListener": [MockFunction],
              "interpolate": [MockFunction],
              "removeAllListeners": [MockFunction],
              "removeListener": [MockFunction],
              "start": [MockFunction],
            },
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </RNGestureHandlerButton>
</RCTView>
`;
